### PR TITLE
Instructions for uninstalling sail-operator

### PR DESCRIPTION
**Pull request by sridhargaddam**

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -25,4 +25,15 @@ tbd
 tbd
 
 ## Uninstalling
-tbd
+
+To uninstall the `sail-operator` from the cluster, you can execute the following command. This will remove all Istio CRDs, the sail-operator deployment, and the associated namespace.
+
+```shell
+make undeploy
```
Comment (dgn) [inline docs/README.md:32]: As this is end user documentation, I think we should not expect installation to have happened from source, so we need to tell people to remove the subscription in OLM and take care of any residual resources (I don't think we have that problem in sail though- so it's probably just the OLM part)

Comment (sridhargaddam) [reply docs/README.md:32]: Sure @dgn. Let me try once with OCP just to be sure that we are not leaving any artifacts. Will update the PR once I have the details.

Comment (sridhargaddam) [reply docs/README.md]: @dgn, after uninstalling the Istio, IstioCNI resources, the sail-operator and the projects (istio-system, istio-cni) the cluster still has the following CRDs.

```
$ oc get crds -A | grep istio
authorizationpolicies.security.istio.io                           2024-06-18T08:27:28Z
destinationrules.networking.istio.io                              2024-06-18T08:27:29Z
envoyfilters.networking.istio.io                                  2024-06-18T08:27:28Z
gateways.networking.istio.io                                      2024-06-18T08:27:28Z
istiocnis.operator.istio.io                                       2024-06-18T08:27:30Z
istiorevisions.operator.istio.io                                  2024-06-18T08:27:28Z
istios.operator.istio.io                                          2024-06-18T08:27:30Z
peerauthentications.security.istio.io                             2024-06-18T08:27:31Z
proxyconfigs.networking.istio.io                                  2024-06-18T08:27:30Z
requestauthentications.security.istio.io                          2024-06-18T08:27:28Z
serviceentries.networking.istio.io                                2024-06-18T08:27:29Z
sidecars.networking.istio.io                                      2024-06-18T08:27:27Z
telemetries.telemetry.istio.io                                    2024-06-18T08:27:28Z
virtualservices.networking.istio.io                               2024-06-18T08:27:28Z
wasmplugins.extensions.istio.io                                   2024-06-18T08:27:30Z
workloadentries.networking.istio.io                               2024-06-18T08:27:30Z
workloadgroups.networking.istio.io                                2024-06-18T08:27:31Z
sgaddam@fedora:~/deployk8s/crc$ 
```

**Comment by sridhargaddam** [comment]
In the current PR, I captured individual steps to uninstall the Istio resources followed by sail-operator. An alternate approach is to select the individual instances of Istio/IstioCNI/IstioRevision as part of uninstalling the sail-operator from the OpenShift console (PSB). Both seem to achieve the same result.

Screenshot from 2024-06-18 17-05-11

**Comment by bmangoen** [comment]
> In the current PR, I captured individual steps to uninstall the Istio resources followed by sail-operator. An alternate approach is to select the Istio/IstioCNI/IstioRevision instances as part of uninstalling the sail-operator from the OpenShift console (PSB). Both seem to achieve the same result.
> 
> Screenshot from 2024-06-18 17-05-11

This is nice! Good to know

File: docs/README.md (hunk 1 above)
Comment (bmangoen) [reply docs/README.md]: Should we add a last step for this cleaning up?

```
kubectl get crd -oname | grep istio.io | xargs kubectl delete
```

Comment (sridhargaddam) [reply docs/README.md]: Sure, we can do that. Thanks.

Comment (dgn) [reply docs/README.md]: hmm it looks like OLM leaves it to the users whether to remove CRDs: https://olm.operatorframework.io/docs/tasks/uninstall-operator/#step-4-deciding-whether-or-not-to-delete-the-crds-and-apiservices

We can reference that and provide a command that removes the CRDs if wanted

**Review by dgn — APPROVED** [review]
Thanks Sridhar, lgtm so far, we might want to reference OLM docs regarding CRD removal but I'll leave that up to you

File: docs/README.md (hunk 1 above)
Comment (sridhargaddam) [reply docs/README.md]: Thanks @dgn and @bmangoen for reviewing. Let me add a reference to the OLM note and the manual command to delete the CRDs.

**Pull request merged**